lib/pages/demo.dart

```diff
@@ -31,19 +31,31 @@ enum _DemoState {
   fullscreen,
 }
 
-class DemoPage extends StatefulWidget {
-  const DemoPage({
+class DemoPage extends StatelessWidget {
+  const DemoPage({Key key, @required this.slug}) : super(key: key);
+
+  final String slug;
+
+  @override
+  Widget build(BuildContext context) {
+    return GalleryDemoPage(demo: slugToDemo(context)[slug]);
```

Comment:
This might be slightly expensive to run on every build. Is there a way this can be refactored so this only happens once?

Reply:
I agree, let me refactor this a bit.